### PR TITLE
Requirements: Add installation methods

### PR DESCRIPTION
**Pull request by Adrianzatreanu**

### CODE REVIEW

**Comment by sils1297** [comment]
reack https://github.com/coala-analyzer/coala/pull/2408/commits/e0c38059e98f0734c56af6da5ad08e4d0948f656

**Comment by sils1297** [comment]
reack https://github.com/coala-analyzer/coala/pull/2408/commits/822d1b249e28cd37a3fb29ec3c501a09039f6209

File: tests/bears/requirements/PipRequirementTest.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import unittest
+import shutil
+from coalib.bears.requirements.PipRequirement import PipRequirement
+
+
+@unittest.skipIf(shutil.which('pip') is None, "Pip is not installed.")
+class PipRequirementTestCase(unittest.TestCase):
+
+    def test_Install_Command_With_Version(self):
```
Comment (sils1297) [inline tests/bears/requirements/PipRequirementTest.py]: we usually use snake case for functions, that means all lower case and separated by underscores - can you make that right? :)

File: coalib/bears/requirements/GemRequirement.py [hunk 2]
```diff
@@ -29,3 +31,31 @@ def __init__(self, package, version="", require=""):
...
+
+    def install_command(self):
+        """
+        Creates the installation command for the instance of the class.
+
+        >>> GemRequirement('rubocop').install_command()
+        'gem install rubocop'
+
+        >>> GemRequirement('scss_lint', '', 'false').install_command()
+        "gem install 'scss_lint', require: false"
```
Comment (sils1297) [inline coalib/bears/requirements/GemRequirement.py]: why does the lower version have quotes

File: coalib/bears/requirements/GemRequirement.py [hunk 3]
```diff
@@ -29,3 +31,31 @@ def __init__(self, package, version="", require=""):
...
+
+        :param return: A string with the installation command.
+        """
+        if self.require:
+            return "gem install '{}', require: {}".format(self.package,
```
Comment (sils1297) [inline coalib/bears/requirements/GemRequirement.py]: I feel like it'd be better to do something like

```
result = 'gem ...'

if ...:
    result += ...

retur result
```

this way you don't duplicate part of the string leadig to the error pointed out above

File: tests/bears/requirements/GemRequirementTest.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import shutil
+import unittest
+from coalib.bears.requirements.GemRequirement import GemRequirement
+
+
+@unittest.skipIf(shutil.which('gem') is None, "Gem is not installed.")
+class GemRequirementTestCase(unittest.TestCase):
+
+    def test_InstalledRequirement(self):
```
Comment (sils1297) [inline tests/bears/requirements/GemRequirementTest.py]: again in this and the following commits, snake case please

File: coalib/bears/requirements/NpmRequirement.py [hunk 5]
```diff
@@ -24,3 +26,30 @@ def __init__(self, package, version=""):
...
+        'npm install alex@2'
+
+        >>> NpmRequirement('alex').install_command()
+        'npm install alex'
+
+        :param return: A string with the installation command.
+        """
+        if self.version:
+            return "npm install {}@{}".format(self.package, self.version)
+        return "npm install {}".format(self.package)
```
Comment (sils1297) [inline coalib/bears/requirements/NpmRequirement.py]: same as with last commit

**Comment by sils1297** [comment]
reack https://github.com/coala-analyzer/coala/pull/2408/commits/3d1f7e60c0f64057a79f1d8244e51fd147add25b https://github.com/coala-analyzer/coala/pull/2408/commits/349e2c15abc1101597a05b9c73d9ca3c170e8d3f

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2408/commits/b0b3d6b36a4741d01b98e8cb4d98f9e761f3b41d https://github.com/coala-analyzer/coala/pull/2408/commits/b7a7fe5e93e7ddb6eafd0de4f79a9ccaf4d179cd https://github.com/coala-analyzer/coala/pull/2408/commits/c8ba0c9255a5a36904bea63a163d4c5c558ba9ba

**Comment by Adrianzatreanu** [comment]
good choice, sir!

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Pull request merged**